spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -99,6 +109,35 @@ public UnicastSendingMessageHandler(String host, int port) {
 		super(host, port);
 		this.mapper.setLengthCheck(false);
 		this.mapper.setAcknowledge(false);
+		this.socketExpression = null;
+	}
+
+	/**
+	 * Basic constructor using external datagram socket for sending outgoing data.
+	 * @param socketExpression socket expression string describing external socket
+	 * @since 4.3
+	 */
+	public UnicastSendingMessageHandler(String socketExpression) {
+		super("", 0);
+		Assert.hasText(socketExpression);
```

Comment:
Would be better to say anything in case of failure: the second `message` argument.